cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -747,7 +747,7 @@ if(NOT ARROW_GIT_ID)
                   OUTPUT_STRIP_TRAILING_WHITESPACE)
 endif()
 if(NOT ARROW_GIT_DESCRIPTION)
-  execute_process(COMMAND "git" "describe" "--tags" "--dirty"
+  execute_process(COMMAND "git" "describe" "--tags"
```

Comment:
Are you sure this fixes the issue reliably? Is it just an implementation detail in git?

Reply:
It's an implementation detail.

The `git describe` document doesn't explain this behavior: https://git-scm.com/docs/git-describe

Here is the implementation with `--dirty`:

https://github.com/git/git/blob/3a06386e314565108ad56a9bdb8f7b80ac52fb69/builtin/describe.c#L669-L697

It touches `.git/index` (`*_index()` functions are used).

Reply:
wow great investigation!